`Fn` implementation for function pointers to method of `~const` bounded trait are not `const`

I tried this code ([playground](https://play.rust-lang.org/?version=nightly&gist=1270f1da3bf2592d31626173f59f453b)):
```rust
#![feature(const_trait_impl)]

use std::marker::Destruct;

const fn f<T, F: ~const Fn() -> T + ~const Destruct>(_: F) {}

const fn g<T: ~const Default>() {
    f(<T as Default>::default);
}
```

I got an compile error:
```
error[E0277]: the trait bound `fn() -> T {<T as Default>::default}: ~const Fn<()>` is not satisfied
 --> src\lib.rs:8:7
  |
8 |     f(<T as Default>::default);
  |     - ^^^^^^^^^^^^^^^^^^^^^^^ expected an `Fn<()>` closure, found `fn() -> T {<T as Default>::default}`
  |     |
  |     required by a bound introduced by this call
  |
  = help: the trait `~const Fn<()>` is not implemented for fn item `fn() -> T {<T as Default>::default}`
note: the trait `Fn<()>` is implemented for `fn() -> T {<T as Default>::default}`, but that implementation is not `const`
 --> src\lib.rs:8:7
  |
8 |     f(<T as Default>::default);
  |       ^^^^^^^^^^^^^^^^^^^^^^^
```

### Meta

`rustc --version --verbose`:
```
rustc 1.67.0-nightly (42325c525 2022-11-11)
binary: rustc
commit-hash: 42325c525b9d3885847a3f803abe53c562d289da
commit-date: 2022-11-11
host: x86_64-pc-windows-msvc
release: 1.67.0-nightly
LLVM version: 15.0.4
```

--- a/tests/ui/traits/const-traits/const-closure-issue-125866-error.rs
+++ b/tests/ui/traits/const-traits/const-closure-issue-125866-error.rs
@@ -1,0 +1,24 @@
+#![allow(incomplete_features)]
+#![feature(const_closures, const_trait_impl)]
+
+const fn create_array<const N: usize>(mut f: impl FnMut(usize) -> u32 + Copy) -> [u32; N] {
+    let mut array = [0; N];
+    let mut i = 0;
+    loop {
+        array[i] = f(i);
+        //~^ ERROR the trait bound `impl FnMut(usize) -> u32 + Copy: [const] FnMut(usize)` is not satisfied [E0277]
+        i += 1;
+        if i == N {
+            break;
+        }
+    }
+    array
+}
+
+fn main() {
+    let x = create_array(const |i| 2 * i as u32);
+    assert_eq!(x, [0, 2, 4, 6, 8]);
+
+    let y = create_array(const |i| 2 * i as u32 + 1);
+    assert_eq!(y, [1, 3, 5, 7, 9]);
+}

--- a/tests/ui/traits/const-traits/const-closure-issue-125866-error.stderr
+++ b/tests/ui/traits/const-traits/const-closure-issue-125866-error.stderr
@@ -1,0 +1,11 @@
+error[E0277]: the trait bound `impl FnMut(usize) -> u32 + Copy: [const] FnMut(usize)` is not satisfied
+  --> $DIR/const-closure-issue-125866-error.rs:8:22
+   |
+LL |         array[i] = f(i);
+   |                    - ^
+   |                    |
+   |                    required by a bound introduced by this call
+
+error: aborting due to 1 previous error
+
+For more information about this error, try `rustc --explain E0277`.

--- a/tests/ui/traits/const-traits/const-closure-issue-125866-pass.rs
+++ b/tests/ui/traits/const-traits/const-closure-issue-125866-pass.rs
@@ -1,0 +1,25 @@
+//@ check-pass
+
+#![allow(incomplete_features)]
+#![feature(const_closures, const_trait_impl)]
+
+const fn create_array<const N: usize>(mut f: impl [const] FnMut(usize) -> u32 + Copy) -> [u32; N] {
+    let mut array = [0; N];
+    let mut i = 0;
+    loop {
+        array[i] = f(i);
+        i += 1;
+        if i == N {
+            break;
+        }
+    }
+    array
+}
+
+fn main() {
+    let x = create_array(const |i| 2 * i as u32);
+    assert_eq!(x, [0, 2, 4, 6, 8]);
+
+    let y = create_array(const |i| 2 * i as u32 + 1);
+    assert_eq!(y, [1, 3, 5, 7, 9]);
+}

--- a/tests/ui/traits/const-traits/const-fn-trait-bound-issue-104314.rs
+++ b/tests/ui/traits/const-traits/const-fn-trait-bound-issue-104314.rs
@@ -1,0 +1,13 @@
+//@ check-pass
+
+#![feature(const_trait_impl, const_destruct, const_clone)]
+
+use std::marker::Destruct;
+
+const fn f<T, F: [const] Fn(&T) -> T + [const] Destruct>(_: F) {}
+
+const fn g<T: [const] Clone>() {
+    f(<T as Clone>::clone);
+}
+
+fn main() {}

